auditlog/models/rule.py

```diff
@@ -240,7 +240,7 @@ def create(self, vals):
         model = self.env["ir.model"].browse(vals["model_id"])
         vals.update({"model_name": model.name, "model_model": model.model})
         new_record = super().create(vals)
-        if new_record._register_hook():
+        if not self.env.context.get("install_module") and new_record._register_hook():
```

Comment:
I am wondering if we should not move the content of `_register_hook` override in a separate method first (we should add a `rules` parameter to this method, if none is passed we search all rules subscribed as of now), then in the `create`/`write` methods we call it with the current rule we want to subscribe.
So we avoid calling `_register_hook` here which is better I think, what do you think?

Reply:
Or your issue here is not `_register_hook` but the call to `signal_changes`? This one should be done in a `after commit` hook to not trigger it in the middle of a transaction :thinking:

Reply:
Good questions! I'll check them when I have time.

Reply:
If you prefer we could merge your version instead (with a comment explaining what is the intent of your check) and postpone a better solution for later (if there is one...).

Reply:
No, I think a better solution is possible and we should go for that straight away.